src/services/suggestionDiagnostics.ts

```diff
@@ -109,7 +113,118 @@ namespace ts {
         }
     }
 
+    function addConvertToAsyncFunctionDiagnostics(node: Node, checker: TypeChecker, diags: DiagnosticWithLocation[]): void {

```

Comment:
change `node` to be `FunctionLikeDeclaration`. you have already asserted that before calling it.